test/server/gameSystems/CaptureSystem.spec.ts

```diff
@@ -0,0 +1,111 @@
+                expect(context.wampa).toBeInZone('groundArena', context.player2);
+                expect(context.wampa.exhausted).toBeTrue();
+            });
+        });
+
+        it('When multiple units are captured, they should all be in the capture zone and all rescued on capture defeat', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['discerning-veteran', 'take-captive'],
+                },
+                player2: {
+                    groundArena: ['wampa', 'atst'],
+                    hand: ['vanquish']
+                }
+            });
+
+            const { context } = contextRef;
+
+            // capture Wampa with Discerning Veteran
+            context.player1.clickCard(context.discerningVeteran);
+            expect(context.player1).toBeAbleToSelectExactly([context.wampa, context.atst]);
+            context.player1.clickCard(context.wampa);
+            expect(context.wampa).toBeCapturedBy(context.discerningVeteran);
+
+            context.player2.passAction();
+
+            // capture AT-ST with Discerning Veteran
+            context.player1.clickCard(context.takeCaptive);
+            expect(context.atst).toBeCapturedBy(context.discerningVeteran);
+            expect(context.wampa).toBeCapturedBy(context.discerningVeteran);
+
+            // defeat Discerning Veteran, both units rescued
+            context.player2.clickCard(context.vanquish);
+            expect(context.atst).toBeInZone('groundArena');
+            expect(context.wampa).toBeInZone('groundArena');
+            expect(context.atst.exhausted).toBeTrue();
+            expect(context.wampa.exhausted).toBeTrue();
+        });
+
+        it('When a unit is captured that has upgrades and its own captured unit, the upgrades are defeated and the unit is rescued', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['discerning-veteran'],
+                    groundArena: ['battlefield-marine']
+                },
+                player2: {
+                    groundArena: [{ card: 'wampa', upgrades: ['academy-training'] }],
+                    hand: ['take-captive']
+                }
+            });
+
+            const { context } = contextRef;
+
+            context.player1.passAction();
+
+            // capture marine with wampa
+            context.player2.clickCard(context.takeCaptive);
+
+            // capture wampa with discerning veteran
+            context.player1.clickCard(context.discerningVeteran);
+
+            expect(context.wampa).toBeCapturedBy(context.discerningVeteran);
+            expect(context.battlefieldMarine).toBeInZone('groundArena');
+            expect(context.battlefieldMarine.exhausted).toBeTrue();
+            expect(context.academyTraining).toBeInZone('discard');
+        });
+    });
+});
```

Comment:
there are few more tests we can add, 

- [ ] for superlaser blast, captured units should not be defeated 
- [ ] defeat a control-taken unit which guard units should return captured unit to the good controllers

Reply:
I added the first test but I don't know what you mean by the second test, could you give an example?

Reply:
sure : player1 captures a unit with take captive, player2 take control of this unit (with palpatine for example), player1 use rivals fall on this unit. the captured unit must return in play on player 2 side